### DISCUSSION TITLE
pytest multithreading using SeleniumBase

### DISCUSSION THREAD

**Discussion by fashionprivate** [pytest-dev/pytest#11723] [Q&A]
Hello,

I'm using pytest in order to manage a multithreading bot and everything works fine.

```python
import pytest
from seleniumbase import BaseCase

def main():
    BaseCase.main(__name__, __file__, "-n4")

if __name__ == "__main__":
    main()

@pytest.mark.parametrize("", [[]] * 20)
def test_multi_threaded(sb):
    sb.open("https://example.com")
   # very long task
```

Now, how I can get the CTRL-C key command in order to shutdown all the tests that run in parallel?

Thanks

**Comment by RonnyPfannschmidt** [comment]
It's unclear what you are asking for

Pytest is unaware of threads

**Reply by fashionprivate** [reply]
Hello,

I try to be more precise: I developed a bot in Python that scrapes a website.
I divided the work into several workers (about 20).
Since the workers can do very heavy processing in terms of resources used, I gave the possibility to stop the bot via CTRL-C.
I would like to safely deallocate the `sb` object after pressing CTRL-C. Is it possible to call `sb.quit()` as soon as the user presses CTRL-C? If so, how can it be implemented?

**Reply by RonnyPfannschmidt** [reply]
The code in the example does something completely different

Unless you get the description of the problem and the terminology used in order, it's absolutely impossible to give reliable and correct answers

**Reply by fashionprivate** [reply]
I corrected the example but I think it's not important the huge task that have to run the workers. 
The important thing is the logic of the bot. As you can see in the example I have 20 workers all doing the same thing. Is it possible to stop all 20 by calling the `quit()` method on the `sb` object?
Is there a specific handler for this?

What I expect is that there is a handler like this.

```python
@pytest.hookimpl(tryfirst=True)
def pytest_keyboard_interrupt(excinfo):
    # i don't have the reference to the sb object
```

But I don't have the reference to the `sb` object to be able to safely release it

**Reply by RonnyPfannschmidt** [reply]
technically its currently not possibly to correctly and reliably pass over a keyboard interrupt to execnet workers due to a long standing bug
im not aware of a easy solution, however the normal session shutdown by the coordinator of pytest should work by default

more context is needed
